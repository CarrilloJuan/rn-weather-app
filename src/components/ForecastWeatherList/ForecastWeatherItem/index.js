import React from 'react';
import {Text, View} from 'react-native';
import PropTypes from 'prop-types';

import WeatherIcon from '../../WeatherIcon';
import styles from './styles';

export default function ForecastWeatherDay({day, icon, temp}) {
  return (
    <View style={styles.container}>
      <View style={{flex: 2}}>
        <Text style={styles.text}>{day}</Text>
      </View>
      <View style={styles.temp}>
        <WeatherIcon icon={icon} style={styles.weatherIcon} />
        <Text style={styles.text}>{temp}ÂºC</Text>
      </View>
    </View>
  );
}

ForecastWeatherDay.propTypes = {
  day: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temp: PropTypes.string.isRequired,
};
