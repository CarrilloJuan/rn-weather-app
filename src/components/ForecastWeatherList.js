import React from 'react';
import {FlatList} from 'react-native';
import ForecastWeatherItem from './ForecastWeatherItem';

const forecastMock = [
  {
    day: 'Mon',
    icon: '01d',
    temp: '26º/12ºC',
  },
  {
    day: 'Tue',
    icon: '03d',
    temp: '27º/11ºC',
  },
  {
    day: 'Wed',
    icon: '02d',
    temp: '26º/14ºC',
  },
  {
    day: 'Thu',
    icon: '10d',
    temp: '23º/12ºC',
  },
  {
    day: 'Fri',
    icon: '10d',
    temp: '15º/9ºC',
  },
];

export default function ForecastWeather() {
  const renderItem = ({item}) => <ForecastWeatherItem {...item} />;

  return (
    <FlatList
      data={forecastMock}
      renderItem={renderItem}
      keyExtractor={(item) => item.day}
    />
  );
}
