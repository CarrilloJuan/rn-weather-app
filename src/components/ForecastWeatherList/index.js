import React from 'react';
import {FlatList} from 'react-native';
import {useSelector, shallowEqual} from 'react-redux';

import ForecastWeatherItem from './ForecastWeatherItem';

export default function ForecastWeather() {
  const {forecast} = useSelector(
    ({weather}) => ({
      forecast: weather.data.forecast,
    }),
    shallowEqual,
  );
  const renderItem = ({item}) => <ForecastWeatherItem {...item} />;
  return (
    <FlatList
      data={forecast}
      renderItem={renderItem}
      keyExtractor={(item) => item.day}
    />
  );
}
