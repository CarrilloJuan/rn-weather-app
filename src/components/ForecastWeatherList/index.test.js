/**
 * @jest-environment jsdom
 */

import React from 'react';
import {shallow, mount} from 'enzyme';
import {useSelector} from 'react-redux';
import ForecastWeatherList from './';

const mockState = {
  weather: {
    data: {
      forecast: [
        {
          day: 'Mon',
          icon: '01d',
          temp: '26º/12ºC',
        },
        {
          day: 'Tue',
          icon: '03d',
          temp: '27º/11ºC',
        },
        {
          day: 'Wed',
          icon: '02d',
          temp: '26º/14ºC',
        },
        {
          day: 'Thu',
          icon: '10d',
          temp: '23º/12ºC',
        },
        {
          day: 'Fri',
          icon: '10d',
          temp: '15º/9ºC',
        },
      ],
    },
  },
};

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  shallowEqual: jest.fn(),
}));

describe('Forecast weather component', () => {
  // Remove warnings when rendering react-native components
  const origConsole = console.error;
  beforeEach(() => {
    console.error = () => {};
  });

  afterEach(() => {
    console.error = origConsole;
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('renders as expected', () => {
    useSelector.mockImplementation((selector) => selector(mockState));
    const wrapper = shallow(<ForecastWeatherList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render five items', () => {
    useSelector.mockImplementation((selector) => selector(mockState));
    const wrapper = mount(<ForecastWeatherList />);

    expect(wrapper.find({testID: 'forecast-item'})).toHaveLength(5);
  });
});
