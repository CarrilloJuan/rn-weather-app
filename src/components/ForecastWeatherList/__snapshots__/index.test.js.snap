// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast weather component renders as expected 1`] = `
<FlatList
  data={
    Array [
      Object {
        "day": "Mon",
        "icon": "01d",
        "temp": "26º/12ºC",
      },
      Object {
        "day": "Tue",
        "icon": "03d",
        "temp": "27º/11ºC",
      },
      Object {
        "day": "Wed",
        "icon": "02d",
        "temp": "26º/14ºC",
      },
      Object {
        "day": "Thu",
        "icon": "10d",
        "temp": "23º/12ºC",
      },
      Object {
        "day": "Fri",
        "icon": "10d",
        "temp": "15º/9ºC",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
